File: app/src/main/java/com/drdisagree/colorblendr/service/ScheduledJobService.java
package com.drdisagree.colorblendr.service;

import android.app.job.JobParameters;
import android.app.job.JobService;
import android.content.Intent;
import android.util.Log;

import androidx.work.Configuration;

public class ScheduledJobService extends JobService {

    private static final String TAG = ScheduledJobService.class.getSimpleName();

    ScheduledJobService() {
        @SuppressWarnings("all") Configuration.Builder builder = new Configuration.Builder();
        builder.setJobSchedulerJobIdRange(0, 1000);
    }

    @Override
    public boolean onStartJob(JobParameters jobParameters) {
        ServiceLauncher serviceLauncher = new ServiceLauncher();
        serviceLauncher.launchService(this);

        return false;
    }

    @Override
    public boolean onStopJob(JobParameters jobParameters) {
        Log.i(TAG, "Stopping job...");

        Intent broadcastIntent = new Intent(getApplicationContext().getPackageName());
        sendBroadcast(broadcastIntent);

        return false;
    }
}

File: app/src/main/java/com/drdisagree/colorblendr/service/ScheduledJobService.kt
package com.drdisagree.colorblendr.service

import android.app.job.JobParameters
import android.app.job.JobService
import android.content.Intent
import android.util.Log
import androidx.work.Configuration

class ScheduledJobService internal constructor() : JobService() {

    init {
        val builder: Configuration.Builder = Configuration.Builder()
        builder.setJobSchedulerJobIdRange(0, 1000)
    }

    override fun onStartJob(jobParameters: JobParameters): Boolean {
        val serviceLauncher = ServiceLauncher()
        serviceLauncher.launchService(this)

        return false
    }

    override fun onStopJob(jobParameters: JobParameters): Boolean {
        Log.i(TAG, "Stopping job...")

        val broadcastIntent = Intent(applicationContext.packageName)
        sendBroadcast(broadcastIntent)

        return false
    }

    companion object {
        private val TAG: String = ScheduledJobService::class.java.simpleName
    }
}